grammar.js

```diff
@@ -1487,7 +1487,8 @@ module.exports = grammar({
       field('right', $._type)
     )),
 
-    _identifier_token: $ => token(seq(optional('@'), /[a-zA-Zα-ωΑ-Ωµ_][a-zA-Zα-ωΑ-Ωµ_0-9]*/)), // identifier_token in Roslyn
+    // Unicode categories: L = Letter, Nd = Decimal_Number, Pc = Connector_Punctuation, Cf = Format, Mn = Nonspacing_Mark, Mc = Spacing_Mark, Me = Enclosing_Mark
+    _identifier_token: $ => token(seq(optional('@'), /[\p{L}_][\p{L}\p{Nd}\p{Pc}\p{Cf}\p{Mn}\p{Mc}\p{Me}]*/)),
```

Comment:
I'm not 100% sure the first letter can be any unicode letter. The page https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names that the first character has to be a _ or "a letter". It then goes on to say the next characters may contain "Unicode letter characters" which seems to be an intentional difference.

I tried `𐐣` in Visual Studio and it's okay after the first letter but not AS the first letter. I think we can either leave the first letter matcher as-is and adopt the rest or just go with what you have here and know we're being more relaxed than the real parser. I can roll with either so it's your call.

e.g.
```csharp
var 𐐣 = 1;
```
Gives an error in Visual Studio 2019 while:

```csharp
var _𐐣 = 1;
```

Does not.